src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -129,14 +129,30 @@ void SettingsLoader::GenerateProfiles()
 {
     const auto executeGenerator = [&](const auto& generator) {
         const auto generatorNamespace = generator.GetNamespace();
+        if (_ignoredNamespaces.count(generatorNamespace))
+        {
+            return;
+        }
 
-        if (!_ignoredNamespaces.count(generatorNamespace))
+        const auto previousSize = inboxSettings.profiles.size();
+
+        try
         {
-            try
+            generator.GenerateProfiles(inboxSettings.profiles);
+        }
+        CATCH_LOG_MSG("Dynamic Profile Namespace: \"%.*s\"", gsl::narrow<int>(generatorNamespace.size()), generatorNamespace.data())
+
+        // If the generator produced some profiles we're going to give them default attributes.
+        // By settings the Origin/Source/etc. here, we deduplicate some code and ensure they aren't missing accidentally.
+        if (inboxSettings.profiles.size() > previousSize)
+        {
+            const winrt::hstring source{ generatorNamespace };
+
+            for (const auto& profile : gsl::span(inboxSettings.profiles).subspan(previousSize))

```

Comment:
thanks for moving this here -- clever way to do it too

Reply:
I felt like it's rather "clever" as well and didn't like it.
But I also felt like it's sufficient so I left it that way.